libs/wingc/src/capture.rs

```diff
@@ -18,41 +17,47 @@ pub struct CaptureDef {
 
 struct Capture {
 	pub object: Symbol,
-	pub def: CaptureDef,
+	pub kind: CaptureKind,
 }
 
-pub type Captures = BTreeMap<String, BTreeSet<CaptureDef>>;
+#[derive(Debug, PartialEq, Eq, Hash, Ord, PartialOrd)]
+pub enum CaptureKind {
+	Resource(CaptureDef),
+	ImmutableData,
+}
+
+pub type Captures = BTreeMap<String, BTreeSet<CaptureKind>>;
 
 fn collect_captures(capture_list: Vec<Capture>) -> Captures {
 	let mut captures: Captures = BTreeMap::new();
 	for capture in capture_list {
 		captures
 			.entry(capture.object.name)
 			.or_insert(BTreeSet::new())
-			.insert(capture.def);
+			.insert(capture.kind);
 	}
 	captures
 }
 
-pub fn scan_for_inflights_in_scope(scope: &Scope) {
```

Comment:
nit: We may be able to avoid passing around `diagnostics` everywhere by associating all of these scanning methods with a single struct like `CaptureScanner` that takes ownership of the diagnostics object, and making each method an implementation on the struct (so they each accept `&mut self` as an argument and can all `self.diagnostics.push(...)` inside).

Might not be a quick refactor, so feel free to skip